core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -436,84 +415,83 @@ public boolean connect(Socket socket) throws IOException {
         }
     }
 
-    /**
-     * Reschedules the {@code TcpSlaveAgentListener} on demand.  Disables itself after running.
-     */
     @Extension
     @Restricted(NoExternalUse.class)
-    public static class TcpSlaveAgentListenerRescheduler extends AperiodicWork {
-        private Thread originThread;
-        private Throwable cause;
-        private long recurrencePeriod = 5000;
-        private boolean isActive;
-
-        public TcpSlaveAgentListenerRescheduler() {
-            isActive = false;
-        }
+    public static final class TcpSlaveAgentListenerRescheduler extends AperiodicWork {
+        private static final long RECURRENCE_PERIOD = 5000L;
+        private static volatile boolean active = false;
+
+        public TcpSlaveAgentListenerRescheduler() {}
 
+        /**
+         * @deprecated use {@link TcpSlaveAgentListenerRescheduler#TcpSlaveAgentListenerRescheduler()}
+         */
+        @Deprecated(forRemoval = true)
```

Comment:
```suggestion
        @Deprecated(forRemoval = true, since = "TODO")
```
We're using Java 11 as baseline, hence we can use the `since` string to provide more insights since when something is deprecated.